gdal/port/cpl_http.cpp

```diff
@@ -514,7 +514,9 @@ double CPLHTTPGetNewRetryDelay(int response_code, double dfOldDelay,
     if( response_code == 429 || response_code == 500 ||
         (response_code >= 502 && response_code <= 504) ||
         // S3 sends some client timeout errors as 400 Client Error
-        (response_code == 400 && pszErrBuf && strstr(pszErrBuf, "RequestTimeout")) )
+        (response_code == 400 && pszErrBuf && strstr(pszErrBuf, "RequestTimeout")) ||
+        pszErrBuf  // any transfer/connection/socket error
```

Comment:
`(response_code == 400 && pszErrBuf && strstr(pszErrBuf, "RequestTimeout")) || pszErrBuf` is equivalent to pszErrBuf. But I'm wondering if pszErrBuf wouldn't be set for 401 Unauthorized errors which we don't want to retry

Reply:
Yeah, this case puzzled me, I was going to look back in Git history.

If _libcurl_ returns a RequestTimeout message (which is `pszErrBuf`), then that almost implies "I started getting a 400 response and eventually curl gave up waiting". Which seems awfully odd server behaviour